pkg/networks/validate.go

```diff
@@ -15,26 +15,47 @@ import (
 func (config *NetworksConfig) Validate() error {
 	// validate all paths.* values
 	paths := reflect.ValueOf(&config.Paths).Elem()
+	pathsMap := make(map[string]string, paths.NumField())
+	var socketVMNetNotFound, vdeVMNetNotFound, vdeSwitchNotFound bool
 	for i := 0; i < paths.NumField(); i++ {
 		// extract YAML name from struct tag; strip options like "omitempty"
 		name := paths.Type().Field(i).Tag.Get("yaml")
 		if i := strings.IndexRune(name, ','); i > -1 {
 			name = name[:i]
 		}
 		path := paths.Field(i).Interface().(string)
+		pathsMap[name] = path
 		// varPath will be created securely, but any existing parent directories must already be secure
 		if name == "varRun" {
 			path = findBaseDirectory(path)
 		}
 		err := validatePath(path, name == "varRun")
 		if err != nil {
-			// sudoers file does not need to exist; otherwise `limactl sudoers` couldn't bootstrap
-			if name == "sudoers" && errors.Is(err, os.ErrNotExist) {
-				continue
+			if errors.Is(err, os.ErrNotExist) {
+				switch name {
+				// sudoers file does not need to exist; otherwise `limactl sudoers` couldn't bootstrap
+				case "sudoers":
+					continue
+				case "socketVMNet":
+					socketVMNetNotFound = true
+					continue
+				case "vdeVMNet":
+					vdeVMNetNotFound = true
+					continue
+				case "vdeSwitch":
+					vdeSwitchNotFound = true
+					continue
+				}
 			}
 			return fmt.Errorf("networks.yaml field `paths.%s` error: %w", name, err)
 		}
 	}
+	if socketVMNetNotFound && vdeVMNetNotFound {
+		return fmt.Errorf("networks.yaml: either %q (`paths.socketVMNet`) or %q (`paths.vdeVMNet`) has to be installed", pathsMap["socketVMNet"], pathsMap["vdeVMNet"])
+	}
+	if socketVMNetNotFound && !vdeVMNetNotFound && vdeSwitchNotFound {
+		return fmt.Errorf("networks.yaml: %q (`paths.vdeVMNet`) requires %q (`paths.vdeSwitch`) to be installed", pathsMap["vdeVMNet"], pathsMap["vdeSwitch"])
+	}
```

Comment:
I've been unable to trigger these errors, both by commenting out paths in `networks.yaml` and by renaming the daemon files under `/opt/vde/bin` and `/opt/socket_vmnet/bin`.

Reply:
When these are uninstalled, limactl in an earlier place:

```console
$ limactl start template://vmnet 
? Creating an instance "vmnet" Proceed with the current configuration
INFO[0000] Starting vde_switch daemon for "shared" network 
FATA[0000] daemon "vde_switch" is not available 
```